src/main/java/software/amazon/awssdk/http/apache/internal/conn/SdkTlsSocketFactory.java

```diff
@@ -50,9 +50,16 @@ public SdkTlsSocketFactory(final SSLContext sslContext, final HostnameVerifier h
     @Override
     protected final void prepareSocket(final SSLSocket socket) {
         // BEGIN GG MODIFICATIONS
-        SSLParameters params = new SSLParameters();
-        params.setApplicationProtocols(new String[]{"x-amzn-http-ca", "http/1.1"});
-        socket.setSSLParameters(params);
+        try {
+            SSLParameters params = new SSLParameters();
+            params.setApplicationProtocols(new String[]{"x-amzn-http-ca", "http/1.1"});
+            socket.setSSLParameters(params);
+        } catch (NoSuchMethodError e) {
+            log.debug(() -> "Unable to configure socket for ALPN. Ports other than 443 may not work.");
```

Comment:
nit: Why debug?

Reply:
This method is called for every single SDK operation that needs a socket, so this would massively pollute the logs if it were any higher.

Reply:
`Ports other than 443 may not work.`
-- Do you mean `may work`?